PaolaSantolloV/react-apprenticeship-capstone1
# feat: add the state management

## Description

This PR contain the Mini-Challenge 4: State Management.

- The AuthProvider was changed to GlobalProvider
- The appearance theme was stored in GlobalProvider
-  The search terms was stored in global provider
- Implement useReducer in GlobalProvider
- Add some test in GlobalProvider

## Type of Change


## How should this be tested?

In your terminal run` npm install `after that `npm run start`, now you can see the page on [http://localhost:3000/](http://localhost:3000/)
To run the test `npm run test`, if you want to watch the coverage `npm run test a -- --coverage`.

## Any background context you want to provide?

On the Login modal you can write wherever username and password, this form isn't validate yet.

## What is the deliverable?

Second deliverable

## Screenshots and links

Netlify link: https://capstone1project-4minichallenge.netlify.app/
Covergare : 74.83%

![Screen Shot 2022-01-13 at 12 07 28](https://user-images.githubusercontent.com/97254325/149385666-b89df010-7f62-41c6-a791-7f251a8fb2ff.png)

## Notes
For this PR I added a simple test, I will continue research what tests could be added on the Global Provider.

## Checklist
